patroni/ha.py

```diff
@@ -1147,7 +1161,8 @@ def _run_cycle(self):
             return 'Error communicating with PostgreSQL. Will try again later'
         finally:
             if not dcs_failed:
-                self.touch_member()
+                if not self.touch_member() and not self.is_leader():
```

Comment:
Hm.. why would we blow up a cached reset_cluster at this moment and not at the point when we actually fail to load the new data at load_cluster_from_dcs (if touch_member fails it should trigger the failed flag and the next load cluster failure will be treated as an exception). Also, if we nevertheless do it here, why won't we do it for the leader according to DCS?

Reply:
If etcd node is partitioned it is still available for non-quorum reads, which we are using for performance, therefore `load_cluster_from_dcs` will never fail.
Writes to the partitioned etcd node will always fail, because they require the quorum.

Reply:
Good point, but which problem are we trying to solve by erasing the cached state for non-leader members? I assume load_cluster_from_dcs will re-read the same (stale) data once again...

Reply:
You are right, this change should be reverted.